crypto/derive-secret/src/lib.rs

```diff
@@ -79,9 +79,12 @@ impl DerivableSecret {
     ///
     /// This is useful to ensure that the same root secret is not reused
     /// across multiple `fedimint-client` instances for different federations.
+    ///
+    /// We reset the level to 0 here since `fedimint-client` expects its root
+    /// secret to be at that level.
```

Comment:
I see that it's hard to assert anything in the client otherwise, still seems quite inconsistent :/ 

I think this is something we could change later on if it's a problem, so good enough.

Reply:
Why do you find it inconsistent?

Reply:
Now the level gets reset by some derivation fn and incremented by others.

Reply:
@elsirion The way I think about it is that "level" has a meaning in a context it is being used. The outer application can use (or not) `level` to track and prevent bugs at it's level, and do as many derivations levels at it wants, but the moment it's passed to `Client`, the `Client` changes the context to its internal one, as the secret and its derivations becomes its internal details.

Reply:
Sure, but even in the outer application doing derivations using the federation id will reset the level right now.

Reply:
@elsirion Oh, so if it's about this particular function doing the resetting vs the caller, then it's a different story.

If we expect some other callers might want to use it, we should just not change level and move resetting the level to some other explicit call.

Or we can consider this function to always just produce 0-level federation root key, and if the caller wants something else, they can write their own function.

Both are reasonable, IMO. I already wanted `pub fn federation_key` to be a `fedimint-client` function, in which case keeping it private and internal would make resettting more OK thing to do. I doubt anyone should need it anyway.

@shaurya947